src/setup-gcloud.ts

```diff
@@ -40,7 +40,36 @@ export const GCLOUD_METRICS_LABEL = 'github-actions-setup-gcloud';
 export async function run(): Promise<void> {
   // Warn if pinned to HEAD
   if (isPinnedToHead()) {
-    core.warning(pinnedToHeadWarning('v0'));
+    // TODO(sethvargo): switch back to warning level after branch is renamed
+    // from "master" -> "main".
+    core.error(pinnedToHeadWarning('v0'));
+
+    // TODO(sethvargo): remove after branch is renamed from "master" -> "main".
+    const envvar =
+      'SETUP_GCLOUD_I_UNDERSTAND_USING_MASTER_WILL_BREAK_MY_WORKFLOW_ON_APRIL_5_2022';
+    if (process.env.GITHUB_ACTION_REF === 'master' && !process.env[envvar]) {
+      throw new Error(
```

Comment:
It's worth noting that this will break all of _our_ workflows too (especially ones in other repositories). We intentionally float our integration tests to HEAD so we catch breaking changes early.

Reply:
Could we check for GITHUB_REPOSITORY_OWNER to prevent this?